### PR TITLE
Updating typedoc readme to be more verbose

### PR DESCRIPTION
**Pull request by alexsb**
High level things about how revisit works with links to documentation

### CODE REVIEW

**Review by bbollen23 — CHANGES_REQUESTED** [review]
Just need to make a fixes

File: typedocReadMe.md [hunk 1]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+                    "id": "signature",
+                    "prompt": "Your signature",
+                    "required": true,
+                    "location": "belowStimulus",
+                    "type": "shortText",
+                    "placeholder": "Please provide your signature"
+                }
+            ]
+        }
+```JSON
```
[the remark this answers is not in the document]
Comment (bbollen23) [reply typedocReadMe.md]: Fixed this

File: typedocReadMe.md [hunk 2]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+                }
+            ]
+        }
+```
+
+For more detailed documentation on the response section, check out the [documentation](/typedoc/index.html#response).
+
+## Base Components and Inheritance
+
+[Base Components](/typedoc/interfaces/StudyConfig.html#basecomponents] can be used to implement inheritance for components. This is often useful if you want to parameterize a component. For example:
```
Comment (bbollen23) [inline typedocReadMe.md:90]:
```suggestion
[Base Components](/typedoc/interfaces/StudyConfig.html#basecomponents) can be used to implement inheritance for components. This is often useful if you want to parameterize a component. For example: 
```

**Review by bbollen23 — CHANGES_REQUESTED** [review]

File: typedocReadMe.md [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
-The Study Configuration file is how we describe all of the information necessary to create a study. In this configuration, we describe the metadata for the study, the configuration of the UI, the set of components, and how we sequence them in the study. You can find the detailed documentation for the study configuration [here](/typedoc/interfaces/StudyConfig.html).
+The technical documentation can be found [here](modules.html). 
+
+# Components
+
+Componets are where study-specific content goes. ReVISit currently supports four types of components: 
+
+* **Markdown Files** contain formatted text, including links, images, embedded videos, etc. They are useful for introductions, consent forms, help pages, etc. 
+* **Images** can be used as stimuli directly. 
+* **HTML Pages** can be used to create custom stimuly, including interactive stimuli developed with JavaScript
```
[the remark this answers is not in the document]
Comment (bbollen23) [reply typedocReadMe.md]: Fixed in [97bb812](https://github.com/revisit-studies/study/pull/344/commits/97bb81206e28892a4322a791dc2d3c14fe868155)

File: typedocReadMe.md [hunk 4]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+The sequence object of the study configuration defines (a) the order particpants see your components and (b) determines which components they see. ReVISit supports sophisticated oredering strategies, interruptions and skip logic. Specificially, revisit supports: 
+
+* **Ordering Strategies:** 
+    * **Fixed** order: participants see the components the way they are defined in the sequence
+    * **Random** order: the order of the components are randomized
+    * **[Latin Square](https://en.wikipedia.org/wiki/Latin_square)**: perumte the order of stimuli but ensure that for different participants each stimuli occurs at each point in the sequnce.
```
[the remark this answers is not in the document]
Comment (bbollen23) [reply typedocReadMe.md]: Changed wording in [bc2ea76](https://github.com/revisit-studies/study/pull/344/commits/bc2ea764e1ddd5eba23cc7f88c117d9ae3b79485)

File: typedocReadMe.md (hunk 2 above)
Comment (bbollen23) [reply typedocReadMe.md]: Fixed in [97bb812](https://github.com/revisit-studies/study/pull/344/commits/97bb81206e28892a4322a791dc2d3c14fe868155)

File: typedocReadMe.md [hunk 5]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+* [Questionnaire](/typedoc/interfaces/QuestionnaireComponent.html), 
+* [React](/typedoc/interfaces/ReactComponent.html), and 
+* [Markdown](/typedoc/interfaces/MarkdownComponent.html).
+
+Each component extends the [BaseIndividualComponent](/typedoc/interfaces/BaseIndividualComponent.html) interface. To add a component to your study (which can be thought of as a "page" of your study), you add a JSON object representing that component to the "components" object with a key which you can define how you would like. Then, the `type` key in that JSON object controls which type of component you are referring to. 
+
+
+### Collecting Responses
+
+Each component has a list of responses which represents a set of questions to ask to the user for that particular component. The user can describe where the question should be displayed in the UI, the type of response input (e.g., a [numerical response](/typedoc/interfaces/NumericalResponse.html), a [dropdown](/typedoc/interfaces/DropdownResponse.html), a [slider](/typedoc/interfaces/SliderResponse.html), and more. Each response interface extends the [BaseResponse](/typedoc/interfaces/BaseResponse.html) interface.
```
[the remark this answers is not in the document]
Comment (bbollen23) [reply typedocReadMe.md:64]: Fixed in [c1061b7](https://github.com/revisit-studies/study/pull/344/commits/c1061b7f61aecfeb493cea979134ef583008cd95)

File: typedocReadMe.md [hunk 6]
```diff
@@ -1,12 +1,40 @@
...
+To create a study with reVISit, you have to create **components** that contain the content of your study, and you have to create the **study configuration (the reVISit Spec)** that controls when and how these components are shown to participants. Here, we will introduce these at a high level and link to complete documentation where appropriate. 
 
-# Study Configuration
 
-The Study Configuration file is how we describe all of the information necessary to create a study. In this configuration, we describe the metadata for the study, the configuration of the UI, the set of components, and how we sequence them in the study. You can find the detailed documentation for the study configuration [here](/typedoc/interfaces/StudyConfig.html).
+The technical documentation can be found [here](modules.html). 
+
+# Components
+
+Components are where study-specific content goes. ReVISit currently supports four types of components:
```
Comment (JackWilb) [inline typedocReadMe.md:11]:
```suggestion
Components are where study-specific content goes. ReVISit currently supports five types of components: 
```

File: typedocReadMe.md [hunk 7]
```diff
@@ -1,12 +1,40 @@
...
+Components are where study-specific content goes. ReVISit currently supports four types of components: 
+
+* **Markdown Files** contain formatted text, including links, images, embedded videos, etc. They are useful for introductions, consent forms, help pages, etc. 
+* **Images** can be used as stimuli directly. 
+* **HTML Pages** can be used to create custom stimuli, including interactive stimuli developed with JavaScript 
+* **React Components** can be used for sophisticated interactive stimuli. In comparison to HTML pages, react components simplify the communication between reVISit and the stimulus. 
+* **Survey Questions** can be used to elicit structured responses from participants.
+
+
+All of these stimuli can (and commonly are) paired with **responses**. Responses are form elements that capture the elicited responses. Survey questions are basically empty components with responses.
```
Comment (JackWilb) [inline typedocReadMe.md:20]:
```suggestion
All of these stimuli can be (and commonly are) paired with **responses**. Responses are form elements that capture the elicited responses. Survey questions are basically empty components with responses. 
```

File: typedocReadMe.md [hunk 8]
```diff
@@ -1,12 +1,40 @@
...
+* **Markdown Files** contain formatted text, including links, images, embedded videos, etc. They are useful for introductions, consent forms, help pages, etc. 
+* **Images** can be used as stimuli directly. 
+* **HTML Pages** can be used to create custom stimuli, including interactive stimuli developed with JavaScript 
+* **React Components** can be used for sophisticated interactive stimuli. In comparison to HTML pages, react components simplify the communication between reVISit and the stimulus. 
+* **Survey Questions** can be used to elicit structured responses from participants.
+
+
+All of these stimuli can (and commonly are) paired with **responses**. Responses are form elements that capture the elicited responses. Survey questions are basically empty components with responses. 
+
+A component is typically defined in the spec, with the text, code, or image included from a file. The only exception are survey questions, which do not need a file.
```
Comment (JackWilb) [inline typedocReadMe.md:22]:
```suggestion
A component is typically defined in the spec, with the text, code, or image included from a file/URL. The only exception are survey questions, which do not need a file/URL.
```

File: typedocReadMe.md [hunk 9]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+* [Markdown](/typedoc/interfaces/MarkdownComponent.html).
+
+Each component has a list of responses which represents a set of questions to ask to the user for that particular component. The user can describe where the question should be displayed in the UI, the instruction for the response, and the type of response input (e.g., a [numerical response](/typedoc/interfaces/NumericalResponse.html), a [dropdown](/typedoc/interfaces/DropdownResponse.html), a [slider](/typedoc/interfaces/SliderResponse.html), etc). Each response interface extends the [BaseResponse](/typedoc/interfaces/BaseResponse.html) interface.
+
+
+### Collecting Responses
+
+Each component has a list of responses which represents a set of questions to ask to the user for that particular component. The user can describe where the question should be displayed in the UI, the type of response input (e.g., a [numerical response](/typedoc/interfaces/NumericalResponse.html), a [dropdown](/typedoc/interfaces/DropdownResponse.html), a [slider](/typedoc/interfaces/SliderResponse.html), and more. Each response interface extends the [BaseResponse](/typedoc/interfaces/BaseResponse.html) interface.
```
Comment (JackWilb) [inline typedocReadMe.md:64]: Duplicated blocks. Maybe we should just pick one or use the first sentence as a segue?

File: typedocReadMe.md [hunk 10]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+The sequence object of the study configuration defines (a) the order participants see your components and (b) determines which components they see. ReVISit supports sophisticated ordering strategies, interruptions and skip logic. Specifically, revisit supports: 
+
+* **Ordering Strategies:** 
+    * **Fixed** order: participants see the components the way they are defined in the sequence
+    * **Random** order: the order of the components are randomized
+    * **[Latin Square](https://en.wikipedia.org/wiki/Latin_square)**: permute the order of stimuli but ensure that for a set of participants, each component occurs at each index an equal amount of times throughout the sequence (e.g. if there are 100 participants and 10 components, each component is seen at each index 10 times).
+* **Sampling:**  `numSamples` draws a given number of items from a block. numSamples can be used in combination with each ordering strategy
```
Comment (JackWilb) [inline typedocReadMe.md:108]:
```suggestion
* **Sampling:**  `numSamples` draws a given number of items from a block. numSamples can be used in combination with each ordering strategy (while preserving ordering guarantees)
```

File: typedocReadMe.md [hunk 11]
```diff
@@ -18,15 +46,92 @@ The study metadata defines elements such as the study title, authors, and descri
...
+* **Interruptions**  can be used to insert breaks and attention checks into a block
+* **Skips** can be used to control flow based on the response to a question.
```
Comment (JackWilb) [inline typedocReadMe.md:110]:
```suggestion
* **Skips** can be used to control flow based on the response to a question or a component block. 
```

**Review by JackWilb — CHANGES_REQUESTED** [review]

File: typedocReadMe.md (hunk 9 above)
Comment (bbollen23) [reply typedocReadMe.md]: Fixed in [4b1bc13](https://github.com/revisit-studies/study/pull/344/commits/4b1bc13396bb5e182335cd4e4c5637c67be01fd0)

**Review by JackWilb — APPROVED** [review]

**Review by bbollen23 — APPROVED** [review]

**Pull request merged**